sqlmesh/core/engine_adapter/base.py

```diff
@@ -187,7 +187,11 @@ def _get_source_queries(
             raise SQLMeshError(
                 "It is expected that if a DataFrame is passed in then columns_to_types is set"
             )
-        if query_or_df.empty:
+
+        df_is_empty = (
+            query_or_df.empty if isinstance(query_or_df, pd.DataFrame) else query_or_df.isEmpty()
```

Comment:
pyspark and snowpark? should we check hasattr instead?

Reply:
Yes the intention was to handle pandas vs snowpark/pyspark differently (should fix CI failures), but I'm actually not sure if `isEmpty()` is a valid snowpark method - I'll double check this. We could also use `hasattr` here, I don't feel strongly about it, only  a gut feeling that it may be more hacky/brittle. Happy to go either way, though.

Reply:
check snowpark it needs to work with that too

Reply:
yep just did, it doesn't support `isEmpty()`, we need to use `count()` for snowpark dataframes

Reply:
you could use count for all?

Reply:
actually that’s bad for spark

Reply:
is there even an issue imfor empty snowpark and pyspark? this should only be an issue for empty pandas dataframes to sql

Reply:
i think you only need to check empty for pandas dataframes